### PR TITLE
Use token based approach for system-agent

### LINKED ISSUE #756: Investigate alternative authentication mechanisms
There are ways to reduce number of resources provisioned by agent for authentication in a Rancher cluster, by employing different authentication methods.

Investigate possible approaches.

Initial set:
- Agent can issue a CSR request, which will be approved by Rancher to approve agent authentication, with a short `spec.expirationSeconds` value to allow revoking access. 
- Agent can connect via a single set of SA, RoleBinding and Role, using certificate and a token.

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
**What this PR does / why we need it**:
Reduce the footprint of the system-agent RBAC.

Per each cluster there will be created:
- 1 system-agent ServiceAccount

Per each plan there will be temporarily created:
- 1 Role with access to all plan secrets for each machine
- 1 Rolebinging for the role and the cluster system-agent ServiceAccount

On plan completion/failure the role and rolebinding will be revoked

Per each machine there will be created:
- 1 Secret for the system-agent authentication, with unique JWT bound to the secret existence in the API server, and a namespace/name pointer to the plan secret
- 1 Secret for the plan execution

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #756 

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits into logical changes
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by Danil-Grigorev** [comment]
To test issuing token we need a change which is not yet released in CR: https://github.com/kubernetes-sigs/controller-runtime/pull/2969

Getting “ fakeSubResourceWriter does not support create for token”

**Review by alexander-demicev — APPROVED** [review]
lgtm, no comments are blocking merging this PR

**Review by salasberryfin — APPROVED** [review]
LGTM but it needs a rebase.

File: exp/etcdrestore/webhooks/rke2config.go [hunk 1]
```diff
@@ -382,25 +315,13 @@ func (r *RKE2ConfigWebhook) createConnectInfoJson(ctx context.Context, rke2Confi
...
-		},
-		Data: map[string][]byte{
-			connectInfoConfigKey: connectInfoConfigJson,
-		},
-	},
-	); err != nil {
-		if !apierrors.IsAlreadyExists(err) {
-			return err
-		}
+	if err := r.Update(ctx, secret); err != nil {
```
[the remark this answers is not in the document]
Comment (Danil-Grigorev) [reply exp/etcdrestore/webhooks/rke2config.go:323]: It is already performed on reconcile level with exponential backoff, making it non-blocking.

File: exp/etcdrestore/controllers/etcdsnapshotrestore_controller_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	serviceAccountNamespace string
+	planSecretName          string
+	serverUrl               string
+	pem                     string
+	systemAgentVersion      string
+	token                   []byte
+)
+
+var _ = Describe("RKE2ConfigWebhook tests", func() {
+	// It("Should create a role with the correct properties", func() {
```
Comment (furkatgofurov7) [inline exp/etcdrestore/controllers/etcdsnapshotrestore_controller_test.go:38]: should the code below removed if commented out?

**Review by furkatgofurov7 — APPROVED** [review]

File: exp/etcdrestore/controllers/etcdsnapshotrestore_controller_test.go (hunk 2 above)
Comment (Danil-Grigorev) [reply exp/etcdrestore/controllers/etcdsnapshotrestore_controller_test.go:38]: This was committed by accident, will remove it. Thanks

**Review by salasberryfin — APPROVED** [review]

**Review by furkatgofurov7 — APPROVED** [review]

**Comment by Danil-Grigorev** [comment]
I’ll try to address the rest of the comments in the followup PRs.

**Pull request merged**